File: scripts/agent-bench/fixture-ts/src/rank.ts
/** Reciprocal-rank fusion constant: rank dampening for the tail. */
export const RRF_K = 60;

/**
 * Blend two ranked lists [id, score] into one, reciprocal-rank style: each list
 * contributes 1/(RRF_K + rank), summed per id, highest first.
 */
export function blendScores(
  lexical: Array<[string, number]>,
  semantic: Array<[string, number]>,
): Array<[string, number]> {
  const fused = new Map<string, number>();
  lexical.forEach(([id], rank) => {
    fused.set(id, (fused.get(id) ?? 0) + 1 / (RRF_K + rank + 1));
  });
  semantic.forEach(([id], rank) => {
    fused.set(id, (fused.get(id) ?? 0) + 1 / (RRF_K + rank + 1));
  });
  return [...fused.entries()].sort((a, b) => b[1] - a[1]);
}

File: scripts/agent-bench/fixture-rust/src/rank.rs
/// Reciprocal-rank fusion constant: rank dampening for the tail.
pub const RRF_K: f32 = 60.0;

/// Blend two ranked lists (id, score) into one, reciprocal-rank style: each list
/// contributes 1/(RRF_K + rank), summed per id, highest first.
pub fn blend_scores(lexical: &[(String, f32)], semantic: &[(String, f32)]) -> Vec<(String, f32)> {
    let mut fused: std::collections::HashMap<String, f32> = std::collections::HashMap::new();
    for (rank, (id, _)) in lexical.iter().enumerate() {
        *fused.entry(id.clone()).or_insert(0.0) += 1.0 / (RRF_K + rank as f32 + 1.0);
    }
    for (rank, (id, _)) in semantic.iter().enumerate() {
        *fused.entry(id.clone()).or_insert(0.0) += 1.0 / (RRF_K + rank as f32 + 1.0);
    }
    let mut out: Vec<(String, f32)> = fused.into_iter().collect();
    out.sort_by(|a, b| b.1.partial_cmp(&a.1).unwrap_or(std::cmp::Ordering::Equal));
    out
}
